src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/Knowncolors.cs

```diff
@@ -277,397 +276,537 @@ static internal string MatchColor(string colorString, out bool isKnownColor, out
         }
 #endif
 
-        /// Return the KnownColor from a color string.  If there's no match, KnownColor.UnknownColor
+        /// <summary>
+        /// Return the <see cref="KnownColor"/> from a color string. If there's no match, returns <see cref="KnownColor.UnknownColor"/>.
+        /// </summary>
+        /// <param name="colorString">The color name to parse from.</param>
+        /// <returns>The parsed <see cref="KnownColor"/> value or <see cref="KnownColor.UnknownColor"/> if no match.</returns>
         internal static KnownColor ColorStringToKnownColor(string colorString)
         {
-            if (null != colorString)
+            if (!string.IsNullOrEmpty(colorString))
             {
-                // We use invariant culture because we don't globalize our color names
-                string colorUpper = colorString.ToUpper(System.Globalization.CultureInfo.InvariantCulture);
-
-                // Use String.Equals because it does explicit equality
-                // StartsWith/EndsWith are culture sensitive and are 4-7 times slower than Equals
-
-                switch (colorUpper.Length)
+                switch (colorString.Length)
                 {
                     case 3:
-                        if (colorUpper.Equals("RED")) return KnownColor.Red;
-                        if (colorUpper.Equals("TAN")) return KnownColor.Tan;
+                        if (colorString.Equals("RED", StringComparison.OrdinalIgnoreCase))
```

Comment:
Might be worth to replace the uppercased strings with their pascal-case variant as it is in the enum.

Opinions?

Reply:
Makes sense to me.

Reply:
Fun fact, I just realized I forgot about the first letter matching in the switch case and it didn't completely cross my mind until now.

Reply:
That's why it's good to have something more than "it builds" in the Testing bit...

Reply:
I usually do, sometimes I even include them like #9361 but this one was just like "oh that's easy", haha.

Reply:
I added it even this time just to be sure haha.